elasticdl/python/master/servicer.py

```diff
@@ -145,3 +146,16 @@ def get_average_task_complete_time(self):
 
     def get_worker_liveness_time(self, worker_id):
         return self._worker_liveness_time[worker_id]
+
+    def get_comm_rank(self, request, _):
+        worker_id = request.worker_id
+        k8s_client = self._instance_manager._k8s_client
+        worker_address_port = k8s_client.get_worker_service_address(worker_id)
+        worker_host = worker_address_port.split(":")[0]
+
+        res = elasticdl_pb2.GetRankResponse()
+        res.rank_id = self._rendezvous_server.get_worker_host_rank(worker_host)
```

Comment:
What will happen if `worker_host` is not in _rendezvous_server's worker host list?
Crash?

Reply:
Yes, we can return -1 to tell the worker to retry.